test/unit/test_quilbase.py

```diff
@@ -491,13 +491,12 @@ def def_frame(
         hardware_object: Optional[str],
         sample_rate: Optional[float],
         center_frequency: Optional[float],
-    ):
-        optional_args = [
-            arg
-            for arg in [direction, initial_frequency, hardware_object, sample_rate, center_frequency]
-            if arg is not None
-        ]
-        return DefFrame(frame, *optional_args)
+        channel_delay: Optional[float],
+    ) -> DefFrame:
+        optional_args = {k: v
+                         for k, v in locals().items()
```

Comment:
this is probably more python magic than we want or need

Reply:
I disenchanted it.